doc/source/contributing.rst

```diff
@@ -4,13 +4,574 @@
+of the docs, greatly reducing the turn-around time for checking your changes.
+You will be prompted to delete `.rst` files that aren't required, since the
+last committed version can always be restored from git.
+
+::
+
+    #omit autosummary and API section
+    python make.py clean
+    python make.py --no-api
+
+    # compile the docs with only a single
+    # section, that which is in indexing.rst
+    python make.py clean
+    python make.py --single indexing
+
+For comparison, a full documentation build may take 10 minutes. a ``-no-api`` build
+may take 3 minutes and a single section may take 15 seconds.  However, subsequent 
+builds only process portions you changed.  Now, open the following file in a web 
+browser to see the full documentation you just built::
+
+    pandas/docs/build/html/index.html
+
+And you'll have the satisfaction of seeing your new and improved documentation!
+
+
+Contributing to the code base
+-----------------------------
+
+.. contents:: Code Base:
+   :local:
+
+Code Standards
+^^^^^^^^^^^^^^
+
+*pandas* uses the `PEP8 <http://www.python.org/dev/peps/pep-0008/>`_ standard.  
+There are several tools to ensure you abide by this standard.
+
+We've written a tool to check that your commits are PEP8 great, `pip install pep8radius <https://github.com/hayd/pep8radius>`_. 
+Look at PEP8 fixes in your branch vs master with::
+
+    pep8radius master --diff` and make these changes with `pep8radius master --diff --in-place`
+
+Alternatively, use `flake8 <http://pypi.python.org/pypi/flake8>`_ tool for checking the style of your code.  
+Additional standards are outlined on the `code style wiki page <https://github.com/pydata/pandas/wiki/Code-Style-and-Conventions>`_.
+
+Please try to maintain backward-compatibility. *Pandas* has lots of users with lots of existing code, so 
+don't break it if at all possible.  If you think breakage is required clearly state why 
+as part of the Pull Request.  Also, be careful when changing method signatures and add 
+deprecation warnings where needed.
+
+Test-driven Development/Writing Code
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*Pandas* is serious about `Test-driven Development (TDD) 
+<http://en.wikipedia.org/wiki/Test-driven_development>`_. 
+This development process "relies on the repetition of a very short development cycle: 
+first the developer writes an (initially failing) automated test case that defines a desired 
+improvement or new function, then produces the minimum amount of code to pass that test." 
+So, before actually writing any code, you should write your tests.  Often the test can be 
+taken from the original GitHub issue.  However, it is always worth considering additional 
+use cases and writing corresponding tests.
+
+Adding tests is one of the most common requests after code is pushed to *pandas*.  It is worth getting
+in the habit of writing tests ahead of time so this is never an issue.
```

Comment:
`It is worth getting in the habit of writing tests ahead of time so this is never an issue.`

That's good advice.